src/API/view/view.controller.js

```diff
@@ -1,25 +1,39 @@
 import ViewModel from './view.model';
 
 export default {
-    query
+    query,
+    getById
 };
 
 function query(req, res) {
     ViewModel.find()
-      .populate([{
-          path: 'mapView',
-          populate: {
-              path: 'sensors.sensor',
-              model: 'Sensor'
-          }
-      }, {
-          path: 'dashboard',
-          populate: {
-              path: 'devices',
-              model: 'Sensor'
-          }
-      }])
+      .populate(getViewPopulationConfig())
       .then(result => {
           res.json(result);
       });
 }
+
+function getById(req, { send }, next) {
+    const { id } = req.params;
+
+    ViewModel.findById(id)
+        .populate(getViewPopulationConfig())
+        .then(send)
+        .catch(next);
+}
+
+function getViewPopulationConfig() {
```

Comment:
In next time will be better to move logic (like population, transformation etc) to service (in our case ```view.service.js```) and controller uses only methods from this service.